### ISSUE TITLE
DOC: explain the meaning of model.fit() history tracking in a distributed scenario

### ISSUE THREAD

**Issue by tylerjereddy** [tensorflow/tensorflow#55366]
## URL(s) with the issue:

Perhaps most closely related to some of these docs pages:
  - https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit
  - https://www.tensorflow.org/tutorials/distribute/multi_worker_with_keras#overview

## Description of issue (what needs changing):

There are fairly substantial docs about distributed training with tensorflow, including docs related to the `MultiWorkerMirroredStrategy`. But, as far as I can tell, reading those docs still doesn't prepare the end user for some of the perils of distributed training. One thing I'd like to see described/explained by an expert, is what it means to track the `history` result of `model.fit()` from many different distributed workers.

For example, what does it mean that the learning curves are different between different workers? Should they be? What about accuracy results exceeding 100 % like in the two-worker example results below? Should we even be looking at these individually?

Node 0 `model.fit()` results:

![bench_epoch_rank_0_hostname_cn662](https://user-images.githubusercontent.com/7903078/160017020-721ca1a4-a28c-48d2-bf43-9dadcc14c6f7.png)

Node 1 `model.fit()` results:

![bench_epoch_rank_1_hostname_cn663](https://user-images.githubusercontent.com/7903078/160017031-3300d4b9-de47-4180-a4ac-a02c18aeb298.png)

Should the results be condensed/reduced somehow? For that matter, is there any troubleshooting documentation on what an accuracy greater than 100 % means?

### Correct links

Not related to links.

### Parameters defined

Not related to parameters.

### Returns defined

Perhaps for `model.fit()` there could be a side-note of some sort on how to interpret `history.history` in distributed scenarios, and what kinds of fluctuations between workers are normal vs. suspicious?

### Raises listed and defined

No exception handling issues reported here.

### Usage example

I don't think I've actually seen a good usage example that shows the distributed `history.history` tracking plots differ between workers?

### Request visuals, if applicable

I could probably provide some sample loss/accuracy plots for distributed cases if that would help.

### Submit a pull request?

Are you planning to also submit a pull request to fix the issue? 

If you can offer some guidance on how this is supposed to work, I think that would be useful enough for me to justify helping improve the docs as well.

**Comment by tylerjereddy** [comment]
To make it a bit clearer why this can be confusing, the **exact** same code on a single MPI rank/node produces what looks to me like fairly sensible ML progress curvess--sure the overfitting is extreme because it is one GPU for a few minutes and a microscopic sub-sample of `imagenet2012`, but nothing nearly as wild as the multi-node/multi-GPU case.

![bench_epoch_rank_0_hostname_cn660](https://user-images.githubusercontent.com/7903078/160023369-aa9c9eec-edfb-4eb9-81bf-ac7a85542682.png)

**Comment by tylerjereddy** [comment]
I also almost wonder if an accuracy > 100 % warrants a warning message with suggestions for improvements/likely fixes, but maybe there are custom/debugging ML scenarios where that is valid.

**Comment by tilakrayal** [comment]
@tylerjereddy ,
Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues).
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
